symbolic-uplc/src/ZkFold/Symbolic/UPLC/Evaluation.hs

```diff
@@ -253,6 +298,28 @@ applyPoly ctx SndPair [arg] = do
   MaybeValue p <- evalArg ctx arg []
   (_, ExValue v) <- asPair (Symbolic.fromJust p)
   return $ MaybeValue (symMaybe p v)
+applyPoly ctx (BPFList ChooseList) (ct:tt:et:args) = let
+    b0 = applyPoly ctx (BPFList NullList) [ct]
+    bt =   evalArg ctx (AThunk . Right $ b0) []
+  in applyPoly ctx IfThenElse ((AThunk . Right $ bt):tt:et:args)
+applyPoly ctx (BPFList MkCons) (x0:xs0:_) = do
```

Comment:
I think we expect no more arguments after `xs0`, so we should raise an error if there are more arguments, not ignore them